File: postman/sdks/Specs SDKs/Orders/cli/sdk/orders/create_order_request.go
package orders

import (
	"encoding/json"
	"example.com/orders/sdk/internal/unmarshal"
)

type CreateOrderRequest struct {
	// Customer placing the order
	CustomerID string  `json:"customerId" xml:"customerId" required:"true"`
	Total      float64 `json:"total" xml:"total" required:"true" min:"0"`
	// Initial order status
	Status *CreateOrderRequestStatus `json:"status,omitempty" xml:"status,omitempty"`
	// ISO 4217 currency code
	Currency *string `json:"currency,omitempty" xml:"currency,omitempty" maxLength:"3" minLength:"3"`
}

func (c CreateOrderRequest) String() string {
	jsonData, err := json.MarshalIndent(c, "", "  ")
	if err != nil {
		return "error converting struct: CreateOrderRequest to string"
	}
	return string(jsonData)
}

func (c *CreateOrderRequest) UnmarshalJSON(data []byte) error {
	if err := unmarshal.ValidateRequiredJSONKeys(data, c); err != nil {
		return err
	}
	type alias CreateOrderRequest
	var tmp alias
	if err := json.Unmarshal(data, &tmp); err != nil {
		return err
	}
	*c = CreateOrderRequest(tmp)
	return nil
}

File: postman/sdks/Specs SDKs/Orders/typescript/src/services/orders/models/create-order-request.ts
import { z } from 'zod';
import { CreateOrderRequestStatus, createOrderRequestStatus } from './create-order-request-status';

/**
 * Zod schema for the CreateOrderRequest model.
 * Defines the structure and validation rules for this data type.
 * This is the shape used in application code - what developers interact with.
 */
export const createOrderRequest = z.lazy(() => {
  return z.object({
    customerId: z.string(),
    total: z.number().gte(0),
    status: createOrderRequestStatus.optional(),
    currency: z.string().min(3).max(3).optional(),
  });
});

/**
 * @typedef {CreateOrderRequest} createOrderRequest
 * @property {string} customerId - Customer placing the order
 * @property {number} total
 * @property {CreateOrderRequestStatus} status - Initial order status
 * @property {string} currency - ISO 4217 currency code
 */
export type CreateOrderRequest = z.infer<typeof createOrderRequest>;

/**
 * Zod schema for mapping API responses to the CreateOrderRequest application shape.
 * Handles any property name transformations from the API schema.
 * If property names match the API schema exactly, this is identical to the application shape.
 */
export const createOrderRequestResponse = z.lazy(() => {
  return z
    .object({
      customerId: z.string(),
      total: z.number().gte(0),
      status: createOrderRequestStatus.optional(),
      currency: z.string().min(3).max(3).optional(),
    })
    .transform((data) => ({
      customerId: data['customerId'],
      total: data['total'],
      status: data['status'],
      currency: data['currency'],
    }));
});

/**
 * Zod schema for mapping the CreateOrderRequest application shape to API requests.
 * Handles any property name transformations required by the API schema.
 * If property names match the API schema exactly, this is identical to the application shape.
 */
export const createOrderRequestRequest = z.lazy(() => {
  return z
    .object({
      customerId: z.string(),
      total: z.number().gte(0),
      status: createOrderRequestStatus.optional(),
      currency: z.string().min(3).max(3).optional(),
    })
    .transform((data) => ({
      customerId: data['customerId'],
      total: data['total'],
      status: data['status'],
      currency: data['currency'],
    }));
});
